File: src/com/rian/difficultycalculator/beatmap/timings/EffectControlPointManager.java
package com.rian.difficultycalculator.beatmap.timings;

/**
 * A manager for effect control points.
 */
public class EffectControlPointManager extends ControlPointManager<EffectControlPoint> {
    public EffectControlPointManager() {
        super(new EffectControlPoint(0, false));
    }

    @Override
    public EffectControlPoint controlPointAt(int time) {
        return binarySearchWithFallback(time);
    }
}

File: src/com/osudroid/beatmaps/timings/EffectControlPointManager.kt
package com.osudroid.beatmaps.timings

/**
 * A manager for [EffectControlPoint]s.
 */
class EffectControlPointManager : ControlPointManager<EffectControlPoint>(
    EffectControlPoint(0.0, false)
) {
    override fun controlPointAt(time: Double) = binarySearchWithFallback(time)
}